src/core/or/sendme.c

```diff
@@ -123,10 +139,17 @@ cell_v1_is_valid(const sendme_cell_t *cell, const circuit_t *circ)
 /* Return true iff the given cell version can be handled or if the minimum
  * accepted version from the consensus is known to us. */
 STATIC bool
-cell_version_is_valid(uint8_t cell_version)
+cell_version_is_valid(uint8_t cell_version, const circuit_t *circ)
 {
   int accept_version = get_accept_min_version();
 
+  /* We lookup a different consensus param when receiving a SENDME on a
+   * circuit that is for a directory request. It is only relevant if we are
+   * the directory server and thus on an OR circuit. */
+  if (CIRCUIT_IS_ORCIRC(circ) && CONST_TO_OR_CIRCUIT(circ)->begin_dir_seen) {
+    accept_version = get_dir_min_version();
```

Comment:
This calculation is expensive enough that maybe we don't want to do it for every sendme?